lib/src/pokemon_tcg.dart

```diff
@@ -21,28 +22,32 @@ class PokemonTcgApi {
   static const _baseUrl = 'https://api.pokemontcg.io/v2';
   static const _setsUrl = '$_baseUrl/sets';
 
-  /// Gets a paginated list of all pokemon cards.
+  /// Gets a paginated list of all pokemon cards by default, with an optional search parameter.
   Future<List<PokemonCard>> getCards({
     int page = 0,
+    String query = '',
   }) async {
     http.Response response;
     if (page == 0) {
       response = await client.get(
-        Uri.parse('$_baseUrl/cards'),
+        Uri.parse('$_baseUrl/cards?q=$query'),
         headers: {
           'x-api-key': apiKey,
         },
       );
     } else {
       response = await client.get(
-        Uri.parse('$_baseUrl/cards?page=$page'),
+        Uri.parse('$_baseUrl/cards?q=$query&page=$page'),
         headers: {
           'x-api-key': apiKey,
         },
       );
     }
 
     JsonMap json = jsonDecode(response.body);
+    if (json['error']) {
```

Comment:
This is not how you check if a property exists in Dart. In fact, this will error out when the result is successful now.

You can replace it with the following:

```suggestion
    if (json.containsKey("error")) {
```